storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2156,11 +2156,103 @@ static bool innodb_init_param(void) {
   return (TRUE);
 }
 
+/** Scan the sdi id from mysql.tables
+@param[in]	name 		Name of table database/name
+@param[out]     sdi_id		id of table
+@param[in]	thd 		THD */
+void get_id_from_mysql_table_scan(std::string name, uint64 *id, THD *thd) {
+  std::string db_name = name.substr(0, name.find(OS_PATH_SEPARATOR));
+
+  std::string table_name = name.substr(name.find(OS_PATH_SEPARATOR) + 1);
+  if (table_name.find("#P#") != std::string::npos ||
+      table_name.find("#p#") != std::string::npos) {
+    table_name.resize(table_name.find('#'));
+  }
```

Comment:
we don't need to look into mysql.table_partitions?

Reply:
no not required. it works fine. we store key_id in mysql.tables only. to be sure,  I tried exchanging p3( since metaata is stored in p1) with t1 table and it works fine.

Reply:
ok, thanks for checking